src/components/policy/policy_external/src/cache_manager.cc

```diff
@@ -1888,13 +1898,19 @@ bool CacheManager::SetPredataPolicy(const std::string& app_id) {
   return true;
 }
 
-bool CacheManager::IsPredataPolicy(const std::string& app_id) {
+bool CacheManager::IsPredataPolicy(const std::string& app_id) const {
   // TODO(AOleynik): Maybe change for comparison with pre_DataConsent
   // permissions or check string value from get_string()
-  policy_table::ApplicationParams& pre_data_app =
-      pt_->policy_table.app_policies_section.apps[kPreDataConsentId];
-  policy_table::ApplicationParams& specific_app =
-      pt_->policy_table.app_policies_section.apps[app_id];
+  if (!IsApplicationRepresented(app_id)) {
+    return false;
+  }
+
+  policy_table::ApplicationPolicies& apps =
```

Comment:
@AByzhynar The use of `operator []` requires that `apps` should not be `const`. The constness may be achieved by using iterators combined with `find()` container method. However this requires a check for existence, which is already done in `IsApplicationRepresented` . So the check should not be performed twice and this implementation was selected. The constness may also be achieved by using the `at()` method - however it is from C++11 standard.